misk-hibernate/src/test/kotlin/misk/hibernate/ReflectionQueryFactoryTest.kt

```diff
@@ -420,6 +420,42 @@ class ReflectionQueryFactoryTest {
     }
   }
 
+  @Test
+  fun deleteFailsWithOrderBy() {
+    transacter.transaction { session ->
+      session.save(DbMovie("Rocky 1", null))
+      session.save(DbMovie("Rocky 2", null))
+      session.save(DbMovie("Rocky 3", null))
+    }
+
+    assertThat(assertFailsWith<IllegalStateException> {
+      transacter.transaction { session ->
+        queryFactory.newQuery<OperatorsMovieQuery>().releaseDateAsc().delete(session)
+      }
+    }).hasMessageContaining("orderBy shouldn't be used for a delete")
+  }
+
+  @Test
+  fun delete() {
+    transacter.transaction { session ->
+      session.save(DbMovie("Jurassic Park", LocalDate.of(1993, 6, 9)))
+      session.save(DbMovie("Rocky", LocalDate.of(1976, 11, 21)))
+      session.save(DbMovie("Star Wars", LocalDate.of(1977, 5, 25)))
+    }
+
+    transacter.transaction { session ->
+      queryFactory.newQuery<OperatorsMovieQuery>()
```

Comment:
Can you `assertThat` the result of the delete is `2`?

Reply:
Done